pkg/controller/endpoints/endpoints_mgr.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/endpoints"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta1"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/v1beta2"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+
+	"fmt"
+	"github.com/golang/glog"
+	"reflect"
+)
+
+func ProbeControllerPlugin() []controller.ControllerPlugin {
+	return []controller.ControllerPlugin{&endpointsManager{}}
+}
+
+type endpointsManager struct {
+	host controller.ControllerHost
+}
+
+const (
+	endpointsManagerName = "kubernetes.io/endpoints-manager"
+)
+
+func (plugin *endpointsManager) Init(host controller.ControllerHost) {
+	plugin.host = host
+}
+
+func (plugin *endpointsManager) Name() string {
+	return endpointsManagerName
+}
+
+func (plugin *endpointsManager) GetDefaultSyncPeriod() time.Duration {
```

Comment:
Minor, but I think this should be `DefaultSyncPeriod` and likewise for the others below that start with `Get`.